### PR TITLE
Add a configuration setting allowing to restrict supported identifier types

### PR DESCRIPTION
**Pull request by vbessonov**
## Description

This PR adds a configuration setting allowing to restrict supported identifier types for ODL 2.x and ProQuest collections:
image

Moved the configuration to its own class so other importers may be able to use it. The importer logic is partly in a `IgnoreIdentifierMixin` and partly in the `OPDS2Importer`. This is  primarily because the `ignored_identifiers` MAY be only part of skipped identifiers and more logic may be added later that may be OPDS2 specific.

NOTE: For release notes, this changes the behaviour of the importers, they will now import all types by default where before they were only importing ISBNs. This is the behaviour we want, but it is good to note in the release notes.

## How Has This Been Tested?

Additional tests. Manually tested in my multiple developer environments and in a deployed environment.

## Checklist:

- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by mdellabitta** [comment]
Just was able to do a successful harvest of DPLA's OPDS2 feed with this branch!

**Comment by RishiDiwanTT** [comment]
@tdilauro @jonathangreen Although all tests pass now, I would like to have some real world data to test against.
A feed that has the previously unsupported identifiers.

Apart from the that, the screenshot in the PR description does not match anything I see while trying to create a new collection, nor does any change in the 2 PRs indicate that this additional setting should be present anywhere. Which worries me

**Comment by jonathangreen** [comment]
You should be able to import this feed locally in your CM: 
http://bookshelf-feed-demo.us-east-1.elasticbeanstalk.com/v1/publications

It contains `urn:uuid:` identifiers that would have not been supported before.

**Comment by jonathangreen** [comment]
@RishiDiwanTT when you configured the collection where you able to configure the supported identifier types?

**Comment by RishiDiwanTT** [comment]
Yes 
image

However I would like to turn off an ID type and try, I couldn't figure out which type `urn:uuid` is in the code (or google).
What value should I remove?

File: api/odl2.py
Comment (jonathangreen) [inline api/odl2.py:56]: This seems unrelated to the code here. Did this perhaps come in accidentally?

File: core/opds2_import.py
Comment (jonathangreen) [inline core/opds2_import.py:118]: Comment seems incorrect. I don't think these settings are scoped to the proquest importer.

File: core/opds2_import.py
Comment (jonathangreen) [inline core/opds2_import.py:862]: Should replace these doc comment types with type annotations.

File: core/opds2_import.py
Comment (jonathangreen) [inline core/opds2_import.py:874]: Would be good to remove this since there is a type annotation for it.

File: core/opds2_import.py
Comment (jonathangreen) [inline core/opds2_import.py:891]: Same comment about the type annotations.

File: core/opds_import.py
Comment (jonathangreen) [inline core/opds_import.py:2079]: We should probably still log something here if we don't have a raw identifier, so there is some evidence in the logs that the identifier is missing.

**Review by jonathangreen — CHANGES_REQUESTED** [review]
Looks good other then the few minor comments that I made. Going to do some testing / importing in a test CM now to see how it works. Will add some more comments once that testing is done.

**Review by jonathangreen — CHANGES_REQUESTED** [review]

**Comment by jonathangreen** [comment]
I was able to set this up, and import the bookshelf feed with UUID identifiers. If I enable URI identifiers it works, and if I disable them, the UUIDs aren't imported. So I believe this is working. Just needs some of the other comments here cleaned up.

File: core/opds_import.py
Comment (RishiDiwanTT) [reply core/opds_import.py:2079]: Had the same thought, however this is what the change was in the PR, I can change it to this

File: api/odl2.py
Comment (RishiDiwanTT) [reply api/odl2.py:56]: Yes it seems they are using LICENSE_FORMAT directly from the ODLImporter class. I will remove both

File: core/opds2_import.py
Comment (RishiDiwanTT) [reply core/opds2_import.py:118]: Actually saw this during the merge and forgot about it later. Fixed.

File: api/odl2.py
Comment (RishiDiwanTT) [reply api/odl2.py]: Made it multi-inheritance, option is back in the admin panel now

File: core/model/constants.py
Comment (RishiDiwanTT) [reply core/model/constants.py:221]: Not a fan of this myself, but kept it as the the original PR has this change.
I cannot find anyplace in the code where it is important for this to be in an Enum form, unless the numbers are stored in the DB and correlated somewhere else

File: api/proquest/importer.py
Comment (RishiDiwanTT) [reply api/proquest/importer.py:731]: Yes, I expected the same and had the same confusion about this (here https://github.com/ThePalaceProject/circulation/pull/37#issuecomment-1029006602)

Let me look into this, now that its confirmed

Comment (RishiDiwanTT) [reply api/proquest/importer.py:731]: Looks like the merge from main overwrote this (or the conflict resolution did).
Could you point me in the direction of a proquest feed I can test with

File: core/opds2_import.py
Comment (RishiDiwanTT) [reply core/opds2_import.py:263]: This will change the behaviour of current ODL2 and OPDS2 collections too. Is that desired?

File: core/opds_import.py
Comment (jonathangreen) [reply core/opds_import.py]: Yes. I think this would be good to change.

File: core/model/constants.py
Comment (jonathangreen) [reply core/model/constants.py:221]: I don't believe they are. I think the enum is here to make it easier to iterate over all the identifier values that we have defined.

File: core/opds2_import.py
Comment (jonathangreen) [reply core/opds2_import.py:263]: Yes. I think this is the desired change. I updated to PR description to reflect this and will resolve this comment.

File: api/proquest/importer.py
Comment (jonathangreen) [reply api/proquest/importer.py:731]: You can ask Tara in slack for the feed we configure, but the proquest feed is a bit hard to work with because they keep it behind a firewall only accessible to our production servers. If you want to modify the code here as best you can to make sure these changes work with proquest, I will test with proquest feed on server before merging.

Comment (RishiDiwanTT) [reply api/proquest/importer.py:731]: Changed, but untested

**Comment by RishiDiwanTT** [comment]
> Where if we were only storing the type of identifier that we are blocking, then we wouldn't need to do this database migration.

We are in agreement, I am going ahead with changing this to a "Ignore Identifier type" change

File: tests/core/test_opds2_import.py
Comment (tdilauro) [inline tests/core/test_opds2_import.py:90]: Minor: We can simplify to `super().setup_method()` here? But we can clean these up with `pyupgrade` linter.

Comment (RishiDiwanTT) [reply tests/core/test_opds2_import.py:90]: I can change this quick, no probkems

Comment (RishiDiwanTT) [reply tests/core/test_opds2_import.py:90]: This is actually everywhere, I'd rather leave it to the pyupgrade tool

**Comment by tdilauro** [comment]
I've tested this locally and will deploy to `gorgon` for pre-merge testing in LYR Reads library collections.

**Comment by tdilauro** [comment]
Did some testing in the AWS environment and this seems to work properly.

**Review by tdilauro — APPROVED** [review]
Looks good!

**Comment by tdilauro** [comment]
Fixed the configuration screen image in the description.

**Pull request merged**
